src/main/java/net/helix/pendulum/service/milestone/impl/MilestoneTrackerImpl.java

```diff
@@ -317,7 +319,7 @@ public boolean processMilestoneCandidate(TransactionViewModel transaction) throw
                                 currentRoundViewModel.store(tangle);
                             }
                             addMilestoneToRoundLog(transaction.getHash(), roundIndex, currentRoundViewModel.size(), validators.size());
-                            //setRoundIndexAndConfirmations(currentRoundViewModel, transaction, roundIndex); // todo: uncomment when confirmation count resolved
+                            setRoundIndexAndConfirmations(currentRoundViewModel, transaction, roundIndex);
```

Comment:
Is this also working as expected now?

Reply:
For any valid milestone, transaction confirmation counter is incremented for referenced tips (and all the other transactions from their bundles), also first time when a tip is processed round index is updated.